oak_restricted_kernel/src/lib.rs

```diff
@@ -387,8 +387,14 @@ pub fn start_kernel(info: &BootParams) -> ! {
                 // We need to load the application binary before we hand the channel over to the
                 // syscalls, which expose it to the user space.
                 info!("Loading application binary...");
-                payload::Application::load_raw(&mut *channel)
-                    .expect("failed to load application binary from channel")
+                let payload = oak_channel::basic_framed::load_raw::<
+                    dyn Channel,
+                    { Size4KiB::SIZE as usize },
+                >(&mut *channel)
+                .expect("failed to load application binary from channel");
```

Comment:
I still think it's a decent to move this bit out of kernel, even if it's no longer needed to remove the dependency on micro_rpc.